docs/installation/server-config/security.rst

```diff
@@ -1,4 +1,225 @@
 .. _ksql-security:
 
-Securing KSQL
--------------
\ No newline at end of file
+Configuring Security for KSQL
+=============
+
+KSQL supports many of the security features of both Apache Kafka and the |sr|.
+
+- KSQL supports Apache Kafka security features such as :ref:`SSL for encryption <kafka_ssl_encryption>`,
+  :ref:`SASL for authentication <kafka_sasl_auth>`, and :ref:`Authorization with ACLs <kafka_authorization>`.
+- KSQL supports :ref:`Schema Registry security features <schemaregistry_security>` such as SSL and SASL.
+
+To configure security for KSQL, add your configuration settings to the ``<path-to-confluent>/etc/ksql/ksql-server.properties``
+file and then :ref:`start the KSQL server <install_ksql-server>` with your configuration file specified.
+
+.. code:: bash
+
+    $ <path-to-confluent>/bin/ksql-server-start <path-to-confluent>/etc/ksql/ksql-server.properties
+
+.. contents:: Table of Contents
+    :local:
+
+Configuring KSQL for |ccloud|
+-----------------------------
+
+You can use KSQL with a Kafka cluster in |ccloud|. For more information, see :ref:`install_ksql-ccloud`.
+
+.. _config-security-ksql-sr:
+
+Configuring KSQL for Secured Confluent Schema Registry
+------------------------------------------------------
+
+The following configuration is connecting KSQL with the Confluent Schema Registry over HTTPS.
+
+#. Specify the HTTPS endpoint in the ``ksql.schema.registry.url`` setting in the
+   KSQL server configuration file:
+
+   .. code:: bash
+
+        ksql.schema.registry.url=https://<host-name-of-schema-registry>:<ssl-port>
+
+#. If the Schema Registry's SSL certificate is not signed by a CA that is recognized by the JVM by default, then you
+   must provide a suitable truststore when starting KSQL via the command line:
+
+   .. code:: bash
+
+      $ KSQL_OPTS="-Djavax.net.ssl.trustStore=<path-to-trust-store> -Djavax.net.ssl.trustStorePassword=<store-password>" ksql-server-start <props>
+
+The exact settings you need depend on what SASL mechanism the Confluent Schema Registry is using is using and how your SSL certificates
+are signed. For more information, see :ref:`schemaregistry_security`.
+
+.. _config-security-kafka:
+
+Configuration KSQL for Secured Apache Kafka clusters
+----------------------------------------------------
+
+To connect to a secured Kafka cluster, Kafka client applications must provide their security credentials. The following
+are common configuration examples.
+
+.. _config-security-ssl-sasl:
+
+----------------------------------------------------------
+Configuring Encrypted Communication between KSQL and Kafka
```

Comment:
This section actually covers authentication more than encryption. My fault, I forgot I still had a code & doc change I haven't submitted. Can you add this text to a new section with this title, and rename this section to `Configuring Kafka Authentication`

```
Configuring Encrypted Communication between KSQL and Kafka
-----------------------------------------------------------------

This configuration enables KSQL to connect to a Kafka cluster over SSL, with a user supplied trust store:

.. code:: bash
    security.protocol=SSL
    ssl.truststore.location=/etc/kafka/secrets/kafka.client.truststore.jks
    ssl.truststore.password=confluent

The exact settings you will need will vary depending on the security settings the Kafka brokers
are using and how your SSL certificates are signed. For full details, and instructions on how to create
a suitable trust store, please refer to the`Security section of the Kafka documentation
<http://kafka.apache.org/documentation.html#security>`__.
```

This section should then have a `.. _config-security-ssl` anchor

Reply:
can some of these links to AK docs actually go to CP security docs (a future PR)? https://docs.confluent.io/current/security.html

Reply:
Sure can.